prog/vnet/aws/vpc_nexus.rb

```diff
@@ -0,0 +1,193 @@
+# frozen_string_literal: true
+
+require "aws-sdk-ec2"
+class Prog::Vnet::Aws::VpcNexus < Prog::Base
+  subject_is :private_subnet
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        register_deadline(nil, 10 * 60)
+        hop_destroy
+      end
+    end
+  end
+
+  label def start
+    PrivateSubnetAwsResource.create_with_id(private_subnet.id)
+    hop_create_vpc
+  end
+
+  label def create_vpc
+    vpc_response = client.describe_vpcs({filters: [{name: "tag:Name", values: [private_subnet.name]}]})
+
+    vpc_id = if vpc_response.vpcs.empty?
+      client.create_vpc({cidr_block: private_subnet.net4.to_s,
+        amazon_provided_ipv_6_cidr_block: true,
+        tag_specifications: Util.aws_tag_specifications("vpc", private_subnet.name)}).vpc.vpc_id
+    else
+      vpc_response.vpcs.first.vpc_id
+    end
+
+    private_subnet.private_subnet_aws_resource.update(vpc_id: vpc_id)
+    hop_wait_vpc_created
+  end
+
+  label def wait_vpc_created
+    vpc = client.describe_vpcs({filters: [{name: "vpc-id", values: [private_subnet.private_subnet_aws_resource.vpc_id]}]}).vpcs[0]
+
+    if vpc.state == "available"
+      client.modify_vpc_attribute({
+        vpc_id: vpc.vpc_id,
+        enable_dns_hostnames: {value: true}
+      })
+
+      security_group_response = begin
+        client.create_security_group({
+          group_name: "aws-#{location.name}-#{private_subnet.ubid}",
+          description: "Security group for aws-#{location.name}-#{private_subnet.ubid}",
+          vpc_id: private_subnet.private_subnet_aws_resource.vpc_id,
+          tag_specifications: Util.aws_tag_specifications("security-group", private_subnet.name)
+        })
+      rescue Aws::EC2::Errors::InvalidGroupDuplicate
+        client.describe_security_groups({filters: [{name: "group-name", values: ["aws-#{location.name}-#{private_subnet.ubid}"]}]}).security_groups[0]
+      end
+
+      private_subnet.private_subnet_aws_resource.update(security_group_id: security_group_response.group_id)
+
+      private_subnet.firewalls.flat_map(&:firewall_rules).each do |firewall_rule|
```

Comment:
Avoid N+1 query issue:

```suggestion
      private_subnet.firewalls(eager: :firewall_rules).flat_map(&:firewall_rules).each do |firewall_rule|
```

Alternatively, add a `PrivateSubnet.many_through_many :firewall_rules` association and use that.